doc/user_guide/document_type_plugin_development_guide.md

```diff
@@ -0,0 +1,156 @@
+If your datatype has only one document per file, simply return `Set.of(YOUR_DOCUMTNT_NODE)`.
+Else it is important to stream the data and not to collect it in a List first. 
+For that implement an custom `Iterator` and `Iterable` and use:
+
+```java
+return StreamSupport.stream(new YOUR_ITERABLE(loadedFile).spliterator(), false);
+```
+
+Don't forget to close the input streams!
+
+## The DataLoader
+
+Next, we implement the `DataLoader`. The `DataLoader` runs the whole pipeline from fetching the data, 
+extracting the selecting properties over mapping the result to Exasol values.
+Luckily most of this implementation is covered in its abstract basis `AbstractDataLoader`.
+
+Create a new class `YOUR_TYPEFilesDataLoader` that extends `AbstractDataLoader<YOUR_TYPEDocumentNodeVisitor>`.
+
+Implement the constructor like:
+
+```java
+public YOUR_TYPEFilesDataLoader(final DocumentFetcher<YOUR_TYPEDocumentNodeVisitor> documentFetcher) {
+        super(documentFetcher);
+}
+```
+
+We will call it later from the `YOUR_TYPEDataLoaderFactory`.
+
+Next implement `getValueExtractorFactory` like:
+
+```java
+@Override
+protected PropertyToColumnValueExtractorFactory<YOUR_TYPEDocumentNodeVisitor> getValueExtractorFactory() {
+    return new YOUR_TYPEPropertyToColumnValueExtractorFactory();
+}
+```
+Note that we return here the `PropertyToColumnValueExtractorFactory` that we defined earlier.
+
+## The FilesDataLoaderFactory
+
+Finally, we add a factory for the newly defined `DataLoader`.
+For that create a new class named `YOUR_TYPEFilesDataLoaderFactory`:
+
+```java
+public class YOUR_TYPEFilesDataLoaderFactory extends AbstractFilesDataLoaderFactory {
+    
+    @Override
+    public List<String> getSupportedFileExtensions() {
+        return List.of(".YOUR_TYPE");
+    }
+
+    @Override
+    protected DataLoader buildSingleDataLoader(final FileLoaderFactory fileLoaderFactory,
+            final SegmentDescription segmentDescription, final String sourceString) {
+        return new YOUR_TYPEFilesDataLoader(
+                new YOUR_TYPEDocumentFetcher(sourceString, segmentDescription, fileLoaderFactory));
+    }
+}
+```
+
+Now we introduce the new `DataLoaderFactory` to the generic adapter. 
+To do so create a file `META_INF/services/com.exasol.adapter.document.files.FilesDataLoaderFactory` with the fully qualified name of your `DataLoaderFactory`. For Example:
+
+```
+com.exasol.adapter.document.files.YOUT_TYPEFilesDataLoaderFactory
+``` 
+
+## Tests
+
+Don't forget to test your dialect.
+Take a look at JSON document type (in this repository), as an example.
+
+## Support
+
+If you need help, feel free to create a GitHub issue in this repository.
+
+## Tell the World
+ 
+When you finished your document type implementation, we are happy to list it on the README of this repository, so that others can find it.
```

Comment:
```suggestion
When you are finished your document type implementation, we are happy to list it on the README of this repository, so that others can find it.
```

Reply:
```suggestion
When you are finished with your document type implementation, we are happy to list it on the README of this repository, so that others can find it.
```